packages/terra-switch/src/Switch.jsx

```diff
@@ -0,0 +1,138 @@
+  OFF: <FormattedMessage id="Terra.switch.switchstatuslabel.off" />,
+});
+
+const propTypes = {
+  /**
+   * Id of the Switch button.
+   */
+  switchId: PropTypes.string.isRequired,
+  /**
+   * Whether or not the Switch is checked ("ON").
+   */
+  checked: PropTypes.bool,
+  /**
+   * Whether or not the Switch is disabled.
+   */
+  disabled: PropTypes.bool,
+  /**
+   * The label text of the Switch component.
+   */
+  labelText: PropTypes.string.isRequired,
+  /**
+   * Id of the Switch label.
+   */
+  labelId: PropTypes.string.isRequired,
+  /**
+   * Callback function when switch value changes from ON / OFF.
+   * Returns Parameters: 1. switch value 2. event.
+   */
+  onChange: PropTypes.func,
+};
+
+const defaultProps = {
+  checked: false,
+  disabled: false,
+  onChange: undefined,
+};
+
+const Switch = (props) => {
+  const {
+    switchId,
+    labelId,
+    checked,
+    disabled,
+    onChange,
+    labelText,
+    ...customProps
+  } = props;
+
+  const sliderButton = useRef();
+
+  const handleOnClick = useCallback(
+    (event) => {
+      // See https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Button#Clicking_and_focus
+      // Button on Firefox, Safari and IE running on OS X does not receive focus when clicked.
+      // This will set focus on the button when clicked.
+      sliderButton.current.focus();
+      if (onChange) {
+        onChange(!checked, event);
+      }
+    },
+    [checked, onChange],
+  );
+
+  const switchClassNames = cx([
+    'switch',
+    { 'is-disabled': disabled },
+  ]);
+
+  const trayClassNames = cx([
+    'tray',
+    { selected: checked },
+    { unselected: !checked },
+    { 'is-disabled': disabled },
+  ]);
+
+  const sliderClassNames = cx([
+    'slider',
+    { 'is-disabled': disabled },
+  ]);
+
+  const statusLabelText = checked ? SWITCH_STATE.ON : SWITCH_STATE.OFF;
+
+  const mainClasses = cx([
+    'switch-wrapper',
+    customProps.className,
+  ]);
+
+  delete customProps.className;
+  return (
+    <div {...customProps} className={mainClasses}>
+      {/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */}
+      <label
+        htmlFor={switchId}
+        onMouseDown={disableFocusStyles}
+        onKeyDown={enableFocusStyles}
+        onBlur={restoreFocusStyles}
+      >
+        <div className={switchClassNames}>
+          <div className={cx('label-container')}>
+            <div id={labelId} className={cx('label-text')}>{labelText}</div>
+            <div className={cx('status-label-text')}>{statusLabelText}</div>
+          </div>
+          <div className={trayClassNames}>
+            <button
+              type="button"
+              id={switchId}
+              disabled={disabled}
+              aria-checked={checked}
+              aria-labelledby={labelId}
+              className={sliderClassNames}
+              role="switch"
+              tabIndex="0"
+              onClick={handleOnClick}
+              data-focus-styles-enabled
```

Comment:
should this be?
```suggestion
              data-focus-styles-enabled="false"
```

Reply:
we need it to be true, To show-up focus styles on tabbing from different element. tabbing from different element will certainly not trigger onKeyDown event to enable focus styles so by default we need it to be true.

Reply:
I"m a little confused, when the page loads and the component does not have focus, why should this be true? Also, when clicking on the switch in the deployment, this value it switched to false. Re-tabbing to the component will switch it to true, but tabbing off does not set it to false. Does `data-focus-styles-enabled` true represent what this managing?

Reply:
Probably i might need to change the name then..! Purpose of adding `data-focus-styles-enabled` was to remove focus styles for mouse interaction and to keep it only for keyboard interactions on switch.

> when the page loads and the component does not have focus, why should this be true?

Since focus styles are being dependent on `data-focus-styles-enabled` attr, if it is set to `false` by default then we won't able to set focus styles when we navigate to switch-component for the first time from other component. this is because `data-focus-styles-enabled` will only be set to true on keyboard events of switch. In these since we are tabbing in from different element to switch keyboard event of switch will not get triggered to enable focus styles. So I am keeping it as true by default.

> when clicking on the switch in the deployment, this value it updated to false

it should be set to false only when switch is accessed using mouse. So whenever we click on switch component `data-focus-styles-enabled` will be set to false to avoid focus styles from getting applied. 

> Re-tabbing to the component will update the value to true, but tabbing off does not set it to false

Reason for not setting `data-focus-styles-enabled` to false on tabbing out is for same reason as above we would not able to set focus styles on tabbing back from other element if it was set to true on tab-out. Since the only time where we want it to be false is only for mouse interactions. we are not changing the value as we tab-out of component.